### PR TITLE
Add 2021-05-03-introducing-prometheus-conformance-program.md

### PR DESCRIPTION
**Pull request by RichiH**

### CODE REVIEW

File: content/blog/2021-05-03-introducing-prometheus-conformance-program.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+title: Introducing the Prometheus Conformance Program
+created_at: 2021-05-03
+kind: article
+author_name: Richard "RichiH" Hartmann
+---
+
+Prometheus is the standard for metric monitoring in the cloud native space and beyond. To ensure interoperability, to protect users from suprises, and to enable more parallel innovation, the Prometheus project is introducing the [Prometheus Conformance Program](https://github.com/cncf/prometheus-conformance) with the help of [CNCF]{https://www.cncf.io/} to certify component compliance and Prometheus compatibility.
+
+The CNCF Governing Board is expected to formally start the program during their next meeting. We invite the wider community to help improve our test in this ramp-up phase.
```
Comment (caniszczyk) [inline content/blog/2021-05-03-introducing-prometheus-conformance-program.md:10]: "formally review and approve the program during..."

Comment (RichiH) [reply content/blog/2021-05-03-introducing-prometheus-conformance-program.md]: Fixed, thanks!

File: content/blog/2021-05-03-introducing-prometheus-conformance-program.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+title: Introducing the Prometheus Conformance Program
+created_at: 2021-05-03
+kind: article
+author_name: Richard "RichiH" Hartmann
+---
+
+Prometheus is the standard for metric monitoring in the cloud native space and beyond. To ensure interoperability, to protect users from suprises, and to enable more parallel innovation, the Prometheus project is introducing the [Prometheus Conformance Program](https://github.com/cncf/prometheus-conformance) with the help of [CNCF]{https://www.cncf.io/} to certify component compliance and Prometheus compatibility.
```
Comment (juliusv) [inline content/blog/2021-05-03-introducing-prometheus-conformance-program.md:8]: `{https://www.cncf.io/}` -> `(https://www.cncf.io/)`

File: content/blog/2021-05-03-introducing-prometheus-conformance-program.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+The CNCF Governing Board is expected to formally review and approve the program during their next meeting. We invite the wider community to help improve our test in this ramp-up phase.
```
Comment (juliusv) [inline content/blog/2021-05-03-introducing-prometheus-conformance-program.md:10]: "improve our test" -> "improve our tests"? or "test suite"?

File: content/blog/2021-05-03-introducing-prometheus-conformance-program.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+With the help of our [extensive and expanding test suite](https://github.com/prometheus/compliance), projects and vendors can determine the compliance to our specifications and test suites.
```
Comment (juliusv) [inline content/blog/2021-05-03-introducing-prometheus-conformance-program.md:12]: "test suite(s)" mentioned twice... how about "the compliance with our specifications and implementations"?

File: content/blog/2021-05-03-introducing-prometheus-conformance-program.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+* Remote Read-Write (fully automated, WIP)
+* OpenMetrics (partially automatic, somewhat complete, will need questionnaire)
+
+We plan to add more components. Tests for Prometheus Remote Read or our data storage/TSDB are likely as next additions. We explicitly invite everyone to extend and improve existing tests, and to submit new ones.
+
+The Prometheus Conformance Program works as follows:
+
+For every component, there will be a mark "foo YYYY-MM compliant", e.g. "OpenMetrics 2021-05 compliant", "PromQL 2021-05 compliant", and "Prometheus Remote Write 2021-05 compliant". Any project or vendor can submit their compliance documentation. Upon reaching 100%, the mark will be granted.
+
+For any complete software, there will be a mark "Prometheus x.y compatible", e.g. "Prometheus 2.26 compatible". Relevant component compliance scores are multiplied. Upon reaching 100%, the mark will be granted.
```
Comment (juliusv) [inline content/blog/2021-05-03-introducing-prometheus-conformance-program.md:25]: "Upon reaching 100%, the mark will be granted."

Note that query tweaks in my PromQL tests (https://promlabs.com/promql-compliance-tests) have been treated separately from the % score so far, and the current 100% score wouldn't be a true 100% when we factor those query tweaks in (which we have to, to be fully compliant). E.g. Grafana Cloud would be failing then due to their timestamp alignment, but this is probably also how it should be unless Prometheus Team wants to grant certain exemptions.

Comment (RichiH) [reply content/blog/2021-05-03-introducing-prometheus-conformance-program.md:25]: Our thinking in OM is to follow the system of MUST, SHOULD, MAY and we chose the same language in the Remote Write spec. This maps 1:1 to if tests are fail, warning, or info level. If timestamp alignment MUST NOT happen, then Grafana Cloud does not pass until it stops aligning timestamps, agreed.

This is why I like normative language: It makes gaps and disagreements visible.

**Review by SuperQ — APPROVED** [review]

**Comment by bwplotka** [comment]
Thanks, LGTM! Merged as requestes on live PromCon 2021 ((:

**Pull request merged**